src/hotspot/share/gc/shenandoah/heuristics/shenandoahHeuristics.hpp

```diff
@@ -69,23 +69,96 @@ class ShenandoahHeuristics : public CHeapObj<mtGC> {
   static const intx Degenerated_Penalty = 10; // how much to penalize average GC duration history on Degenerated GC
   static const intx Full_Penalty        = 20; // how much to penalize average GC duration history on Full GC
 
+#ifdef ASSERT
+  enum UnionTag {
+    is_uninitialized, is_garbage, is_live_data
+  };
+#endif
+
 protected:
-  typedef struct {
+  static const uint Moving_Average_Samples = 10; // Number of samples to store in moving averages
```

Comment:
I've never seen that style for constants in HotSpot before. I'd either make it MOVING_AVERAGE_SAMPLES or MovingAverageSamples. I think the latter is more prevalent, but I am not certain.

Reply:
Oh but I see that it's used in this source file. Hmm... So maybe stick with it for now and make a follow-up change?

Reply:
Yes. I agree this is idiosyncratic formatting for the rest of HotSpot, but it is consistent with pre-existing constants in this file:
```C
  static const intx Concurrent_Adjust   = -1; // recover from penalties
  static const intx Degenerated_Penalty = 10; // how much to penalize average GC duration history on Degenerated GC
  static const intx Full_Penalty        = 20; // how much to penalize average GC duration history on Full GC
```